app/sample/ext_frame/rx_st20_pipeline_dyn_ext_frame_sample.c

```diff
@@ -119,18 +121,26 @@ static void rx_st20p_consume_frame(struct rx_st20p_sample_ctx* s,
   s->fb_recv++;
 }
 
+static void wait_frame_available(struct rx_st20p_sample_ctx* s) {
+  st_pthread_mutex_lock(&s->wake_mutex);
+  while (!s->frame_available && !s->stop) {
+    st_pthread_cond_wait(&s->wake_cond, &s->wake_mutex);
```

Comment:
Looping over ```st_pthread_cond_wait()``` is dangerous

Reply:
What do you mean dangerous? It is how it is done even in [manual](https://www.man7.org/linux/man-pages/man3/pthread_cond_wait.3.html)

I'm here trying to fix error from Coverity which says "wait_not_in_locked_loop: A wait is performed without a loop. If there is a spurious wakeup, the condition may not be satisfied"
The issue is that without the loop this can be woken up spuriously so... I added the loop.

Reply:
```
The problem of spurious wakeup can be exacerbated on multiprocessor systems. When several threads are waiting on a single condition variable, the system may decide to wake all threads up when it's signaled
```
How many threads are waiting for wake up ? I believe only one ?
I understand that cv can return from a wait even if not signaled, but this does not happen on linux
```
The Linux p-thread implementation of condition variables guarantees that it will not do that [wake up]
```

Reply:
```
Spurious wakeups from the pthread_cond_clockwait(), pthread_cond_timedwait(), or pthread_cond_wait() functions may occur. 
Since the return from pthread_cond_clockwait(), pthread_cond_timedwait(), or pthread_cond_wait() does not imply anything about the value of this predicate, the predicate should be re-evaluated upon such return.
```
source: [POSIX spec](https://pubs.opengroup.org/onlinepubs/9799919799/)

I'm having trouble finding good evidence that it works differently on Linux, but anyway, I added an explicit variable to improve the readability of the code. You're suggesting that my implementation isn't more readable, so because I would have to change it in many other places I'm giving up and leaving it as is.

After thinking more about it, there should not be any concurrency issues